src/planners/ompl_planner.cpp

```diff
@@ -0,0 +1,106 @@
+#include "ros/ros.h"
+#include "planners/ompl_planner.h"
+
+#include "ompl/base/StateSpace.h"
+#include "ompl/base/spaces/RealVectorStateSpace.h"
+#include "ompl/base/spaces/SO2StateSpace.h"
+
+namespace ob = ompl::base;
+namespace og = ompl::geometric;
+
+using namespace std;
+using namespace krssg_ssl_msgs;
+
+
+Planner::Planner(krssg_ssl_msgs::path_data::Request &request) {
+	start_point  = request.botPos;
+	final_point  = request.targetPos;
+	numObstacles = request.obs_size;
+	botID        = request.botID;
+
+	for(int obs=0; obs<numObstacles; ++obs) {
+		obstacles.push_back(request.obs_vec[obs]);
+	}
+}
+
+Planner::~Planner() {
+
+}
+
+bool Planner::isStateValid(const ob::State *state) {
+  const ob::SE2StateSpace::StateType *state_2d= state->as<ob::SE2StateSpace::StateType>();
+  const double &x(state_2d->getX()), &y(state_2d->getY());
+
+  for (int i = 0; i < numObstacles; ++i){
+    if (sqrt(pow((obstacles[i].x - x), 2)+pow((obstacles[i].y - y), 2)) <= 10.0){
```

Comment:
In fact this entire function will be rewritten once we add the functionality of `LSTM`.